datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/mirrormaker/TestKafkaMirrorMakerConnectorTask.java

```diff
@@ -325,6 +333,109 @@ public void testPartitionManagedLockReleaseOnConsumerCloseException() throws Exc
+    boolean postShutdownHookExceptionCaught;
+
+    /**
+     * Constructor for TestInterruptKafkaMirrorMakerConnectorTask
+     * @param config Task configuration properties
+     * @param task Datastream task
+     * @param connectorName Connector name
+     * @param isFlushlessModeEnabled true if {@value KafkaMirrorMakerConnector#IS_FLUSHLESS_MODE_ENABLED}
+     *                               is set to true for the connector identified with {@code connectorName}
+     * @param groupIdConstructor Kafka consumer group ID constructor
+     */
+    public KafkaMirrorMakerConnectorTaskTest(KafkaBasedConnectorConfig config, DatastreamTask task,
+        String connectorName, boolean isFlushlessModeEnabled, GroupIdConstructor groupIdConstructor) {
+      super(config, task, connectorName, isFlushlessModeEnabled, groupIdConstructor);
+    }
+
+    @Override
+    protected void postShutdownHook() {
+      try {
+        super.postShutdownHook();
+      } catch (Exception e) {
+        postShutdownHookExceptionCaught = true;
+      }
+    }
+
+    boolean isPostShutdownHookExceptionCaught() {
+      return postShutdownHookExceptionCaught;
+    }
+  }
+
+  @Test
+  public void testPartitionManagedLockReleaseOnInterruptException() throws InterruptedException {
+    Datastream datastream = KafkaMirrorMakerConnectorTestUtils.createDatastream("pizzaStream", _broker, "\\w+Pizza");
+    DatastreamTaskImpl task = new DatastreamTaskImpl(Collections.singletonList(datastream));
+    DatastreamEventProducer mockDatastreamEventProducer = mock(DatastreamEventProducer.class);
+    doThrow(InterruptException.class).when(mockDatastreamEventProducer).flush();
+    doNothing().when(mockDatastreamEventProducer).send(any(DatastreamProducerRecord.class), any(SendCallback.class));
+    task.setEventProducer(mockDatastreamEventProducer);
+
+    KafkaBasedConnectorConfig connectorConfig = new KafkaBasedConnectorConfigBuilder()
+        .setConsumerFactory(new LiKafkaConsumerFactory())
+        .setCommitIntervalMillis(10000)
+        .setEnablePartitionManaged(true)
+        .build();
+
+    ZkAdapter zkAdatper = new ZkAdapter(_kafkaCluster.getZkConnection(), "testCluster", null,
+        ZkClient.DEFAULT_SESSION_TIMEOUT, ZkClient.DEFAULT_CONNECTION_TIMEOUT, null);
+    task.setZkAdapter(zkAdatper);
+    zkAdatper.connect();
+
+    KafkaMirrorMakerConnectorTaskTest connectorTask = new KafkaMirrorMakerConnectorTaskTest(connectorConfig, task, "",
+        false, new KafkaMirrorMakerGroupIdConstructor(false, "testCluster"));
+    Thread connectorThread = new Thread(connectorTask, "connector thread");
+    connectorThread.setDaemon(true);
+    AtomicReference<Throwable> uncaughtException = new AtomicReference<>();
+    connectorThread.setUncaughtExceptionHandler((t, e) -> uncaughtException.set(e));
+    connectorThread.start();
+    if (!connectorTask.awaitStart(60, TimeUnit.SECONDS)) {
+      Assert.fail("connector did not start within timeout");
+    }
```

Comment:
Would it better if we replaced this bit with 
```java
Thread connectorThread = KafkaMirrorMakerConnectorTestUtils.runKafkaMirrorMakerConnectorTask(connectorTask);
```
After changing that method to have it return the thread it creates?

Reply:
done